Add note about values.yaml mismatch prevention (#3197)
grafana/mimir: operations/helm/charts/mimir-distributed/CHANGELOG.md

* [ENHANCEMENT] Metamonitoring: If enabled and no URL is configured, then metamonitoring metrics will be sent to
  Mimir under the `metamonitoring` tenant; this enhancement does not apply to GEM. #3176

> * [ENHANCEMENT] Add notice on documentation to prevent user using mismatches helm values.yaml file. #3197

Reviewer: ```suggestion
* [ENHANCEMENT] Documented how to prevent a user from using a mismatched Helm chart `values.yaml` file. #3197
```
